### DISCUSSION TITLE
using axmol-migrate

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1905] [Q&A]
How to use the this migration tool axmol-migrate please guide me in basic simple way.. I'm not a pro so the guidelines are so confusing to me.. 
After cloning the repo.. I went to `axmol-migrate` and ran `pwsh ./build.ps1 -xb "'--config', 'Release'"`

I don't understand this step on that guide.. 

> migrate c++ code: pwsh .\build_x64\axmol-migrate source --fuzzy --source-dir <path/to/your/project/>

there is no .\build_x64\axmol-migrate folder 

How exactly should I run the command ? Sorry if I may sound too naive.

**Comment by smilediver** [comment]
Are there any other `build_XXX` like folders? Btw, the docs for it are a bit outdated. If you figure out how to run the command, then run it without the args to see usage help.

**Reply by meehjeevan** [reply]
No, there is no folder with build_xxx, 
I think this command `pwsh ./build.ps1 -xb "'--config', 'Release'"` should generate that build_xxx folder, but I don't have the latest xcode installed, so may be the setup didn't worked well. 
Is there any way I can skip the xcode part? as I'm using old mac (10.15.7) I can't install the xcode higher than 12.4.

**Reply by smilediver** [reply]
You could try using Ninja to build, the commands would be something like (typing out of my head, so might contain errors):
```
brew install ninja
cmake -G Ninja -B build
cd build
ninja
```
These would install Ninja build tool, then use cmake to generate Ninja project, then you run ninja to build the tool. You don't really need `axmol-migrate`, you could fix your project by hands in the worst case scenario.

**Reply by meehjeevan** [reply]
I got this error:
```
ParserError: 
Line |
  12 |  … �H��h���H�H�BH9�sIH������H�CH��p���H��x���H�H�W�)�p���Hǅ� …
     |                                                                ~
     | Unexpected token ')' in expression or statement.

```

**Reply by rh101** [reply]
Not sure what that error is, but just know that you do not need to use that tool at all, and even if it did work, it would only do a partial job, and you would still need to manually update the app to work with Axmol.

There is no point wasting time on using the axmol-migrate tool if it's not working for you. Simply migrate the app manually; it won't take you long.
